recipe/meta.yaml

```diff
@@ -30,10 +30,20 @@ requirements:
     - defusedxml
 
 test:
+  requires:
+    - git
   imports:
     - cairosvg
   commands:
     - cairosvg --help
+    - git clone --single-branch --branch {{ version }} https://github.com/Kozea/CairoSVG.git
```

Comment:
so, if we need these files for testing then maybe the best thing would be perform the whole install from the github source rather than pypi, downloading the source repo in the `source` section above and bringing the relevant test files in the `test` section with the [`source_files` directive](https://conda.io/projects/conda-build/en/latest/resources/define-metadata.html#source-files).

Reply:
Two things:

1. I tried to install from GitHub source, but CairoSVG has a submodule at /test_non_regression/cairosvg_reference (https://github.com/Kozea/CairoSVG/blob/7520bf6c27848333d78c2e94889380c4e52188c8/.gitmodules#L3) that just refers to a previous commit of CairoSVG.  I think `conda build` attempts to pull that submodule _and all of its submodules_ recursively.  It doesn't reach the bottom of that stack before the VM runs out of disk space.  Or it really does find the bottom but then there's no space left to generate the test outputs.  See the [build log](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=76173&view=logs&jobId=1bf226d3-0e2f-52d8-fa93-7d9e633347b3&taskId=ec8d466d-e3b4-5115-4f06-222398f91e6c&lineStart=398&lineEnd=399&colStart=1&colEnd=1) for ada89c8 if you are curious.  Might be something to report to the core team.
- [ ] I'll see if the `git_depth` directive blocks it, but I assumed that was a commit history depth, not a submodule depth.

2. I'm pulling from the GitHub source due to your request [here](https://github.com/conda-forge/cairosvg-feedstock/pull/19#issuecomment-534181406), quoted below.  Did you mean that we could just copy a file from the CairoSVG repo to this one?  Alternatively, I could also try to just `wget` one file for that one specific issue.
> ...it would be desirable to have a test svg file from the official repo without adding a custom file to the feedstock.

I'm testing all of the files because I didn't really want to get in the business of picking individual files that are good enough for conda-forge.  Then we would want to track why some files were picked and others were excluded... sounds like more work than just testing all of them.  I would be open to just picking ones that get reported to the feedstock, though.  Let me know if you have a preference for `wget` one file, `git clone` and only use one file, or just placing the one file in the feedstock repo.  In the interim:
- [ ] I will check if the file was really a problem only for Windows, or if it was a problem for 2.3.0 generally.

The source repo itself seems to have the perfect solution, but it requires pytest-isort.  I couldn't get it going on my Windows machine.  Maybe it will work in the conda forge anvil, but it wouldn't work if the build ever gets run on a Windows VM:

> If you want to check that CairoSVG works on your computer, you should not launch the tests here, you should instead launch tests in `cairosvg/test_api.py`:
> 
> `pytest cairosvg/test_api.py`

Reply:
All the files can be rendered now without issue.  It was actually a problem with where the working directory is when `cairosvg` is called.  Works on my Windows machine, too.  I think this is a reasonable subset of full testing:  It's not `./setup.py test`, and it works without a lot of manual lifting.  Just "attempt to convert every .svg file in the test folder to a .png file, and check that a non-empty .png file was generated in the attempt".

Reply:
- [x] I will check if the file was really a problem only for Windows, or if it was a problem for 2.3.0 generally.

It really is just a problem for windows.  2.3.0 passes the aggressive rasterize-all-the-files test in my WSL environment.

- [x] I'll see if the git_depth directive blocks it, but I assumed that was a commit history depth, not a submodule depth.

It does not.  I'll report an issue to conda-build.  Until then, I think we have to use git-clone in the test commands section.